lib/synapse/config_generator/haproxy.rb

```diff
@@ -1055,9 +1072,28 @@ def generate_backend_stanza(watcher, config)
             @restart_required = true
           end
         end
+
         backends[backend_name] = backend.merge('enabled' => true)
+
+        # If the the registry defines the haproxy_server_id that must be preferred.
+        server_opts = (backend['haproxy_server_options'] || 'no match').split(' ')
+        @server_id_map[watcher.name][backend_name] = server_opts[server_opts.index('id') + 1].to_i if server_opts.include?('id')
+        # This is going to be an issue ... and haproxy will fail if this happens ... oh well
```

Comment:
thinking about this more, as long as we always choose the ``haproxy_server_options`` value over the ``haproxy_server_id`` this won't conflict, we'll just always use the haproxy_server_id value.

I was referring to the conflict when a server defines two ids (we do it and then later the server_options do it). But as long as we always default to the id in ``haproxy_server_options`` I think we're good.